lib/adal.js

```diff
@@ -709,28 +672,28 @@ AuthenticationContext.prototype.saveTokenFromHash = function (requestInfo) {
             if (requestInfo.parameters.hasOwnProperty(this.CONSTANTS.SESSION_STATE)) {
                 this._saveItem(this.CONSTANTS.STORAGE.SESSION_STATE, requestInfo.parameters[this.CONSTANTS.SESSION_STATE]);
             }
-            
+
             var keys, resource;
-            
+
             if (requestInfo.parameters.hasOwnProperty(this.CONSTANTS.ACCESS_TOKEN)) {
                 this.info('Fragment has access token');
                 // default resource
                 resource = this.config.loginResource;
+                if (requestInfo.requestType === this.REQUEST_TYPE.RENEW_TOKEN) {
```

Comment:
I'm thinking why we have this check here? if the fragment has access token, it can only be renew token request. If so, we should always be able tell the resource based on state response, then why setting the default resource?